### PR TITLE
[PT FE] Make NodeContext constant inside conversion rules

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Refactor NodeContext to be const inside conversion rules*

### Tickets:
 - *None*

### CODE REVIEW

File: src/frontends/pytorch/include/openvino/frontend/pytorch/frontend.hpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ class PYTORCH_API FrontEnd : public ov::frontend::FrontEnd {
     bool supported_impl(const std::vector<ov::Any>& variants) const override;
     ov::frontend::InputModel::Ptr load_impl(const std::vector<ov::Any>& variants) const override;
 
-    std::map<std::string, PytorchCreatorFunction> m_op_translators;
+    std::map<std::string, CreatorFunction> m_op_translators;
```
Comment (rkazants) [inline src/frontends/pytorch/include/openvino/frontend/pytorch/frontend.hpp:63]: `unordered_map`?

File: src/frontends/pytorch/include/openvino/frontend/pytorch/node_context.hpp [hunk 2]
```diff
@@ -130,22 +128,22 @@ class NodeContext : public frontend::NodeContext {
...
     std::shared_ptr<TorchDecoder> m_decoder;
-    std::set<size_t> m_mutated_tensors;
-    TensorMap* m_tensor_map;
     const TensorMap& m_ext_tensor_map;
-    ParameterVector* m_external_parameters;
-    TranslateSession* m_translate_session;
+    TensorMap* m_tensor_map = nullptr;
+    ParameterVector* m_external_parameters = nullptr;
+    std::set<size_t>* m_mutated_tensors = nullptr;
+    TranslateSession* m_translate_session = nullptr;
```
Comment (rkazants) [inline src/frontends/pytorch/include/openvino/frontend/pytorch/node_context.hpp:141]: can we use smart pointers please? It will be more safe.

**Review by rkazants — APPROVED** [review]
left some comments for improvement

File: src/frontends/pytorch/include/openvino/frontend/pytorch/frontend.hpp (hunk 1 above)
Comment (mvafin) [reply src/frontends/pytorch/include/openvino/frontend/pytorch/frontend.hpp:63]: We have map in all frontends, we could change to unordered_map, but this is out of scope of this PR

Comment (rkazants) [reply src/frontends/pytorch/include/openvino/frontend/pytorch/frontend.hpp:63]: np

**Review by eaidova — APPROVED** [review]

File: src/frontends/pytorch/include/openvino/frontend/pytorch/node_context.hpp (hunk 2 above)
Comment (mvafin) [reply src/frontends/pytorch/include/openvino/frontend/pytorch/node_context.hpp]: done

**Pull request merged**